packages/ag-charts-enterprise/src/series/treemap/treemapSeries.ts

```diff
@@ -410,27 +437,71 @@ export class TreemapSeries extends _ModuleSupport.HierarchySeries<TreemapNodeDat
             return;
         }
 
-        const { data, sizeKey, labelKey, colorKey, colorDomain, colorRange, groupFill } = this;
-        const labelFormatter = this.labels.formatter;
+        const {
+            data,
+            colorKey,
+            groupLabelKey,
+            tileLabelKey,
+            tileSecondaryLabelKey,
+            sizeKey,
+            colorDomain,
+            colorRange,
+            tile,
+            group,
+        } = this;
+        const { fill: groupFill } = group;
 
         const colorScale = new ColorScale();
         colorScale.domain = colorDomain;
         colorScale.range = colorRange;
         colorScale.update();
 
         const createTreeNodeDatum = (datum: TreeDatum, depth = 0, parent?: TreemapNodeDatum) => {
-            let label;
-            if (labelFormatter) {
-                label = this.ctx.callbackCache.call(labelFormatter, {
-                    datum,
-                    seriesId: this.id,
-                    value: null,
-                });
-            }
-            label ??= labelKey ? datum[labelKey] ?? '' : '';
-            let colorScaleValue = colorKey ? datum[colorKey] ?? depth : depth;
+            const isLeaf = datum.children == null || datum.children.length === 0;
+
+            const label =
+                isLeaf && tileLabelKey != null
+                    ? this.getLabelText(tile.label, {
+                          datum: datum.datum,
+                          depth: datum.depth,
+                          parent: datum.parent?.datum,
+                          colorKey,
+                          groupLabelKey,
+                          tileLabelKey,
+                          tileSecondaryLabelKey,
+                          sizeKey,
+                          value: datum[tileLabelKey] ?? '',
+                      })
+                    : !isLeaf && groupLabelKey != null
+                    ? this.getLabelText(group.label, {
+                          datum: datum.datum,
+                          depth: datum.depth,
+                          parent: datum.parent?.datum,
+                          colorKey,
+                          groupLabelKey,
+                          tileLabelKey,
+                          tileSecondaryLabelKey,
+                          sizeKey,
+                          value: datum[groupLabelKey] ?? '',
+                      })
+                    : undefined;
```

Comment:
Nit: For non-trivial expressions, we prefer `if ... else` rather than ternary expressions for readability - would you mind reworking these statements a little?

(SonarCloud will flag these in its post-commit report)